code/modules/research/server.dm

```diff
@@ -187,3 +196,112 @@
+
+	SSresearch.master_servers -= src
+
+	return ..()
+
+/obj/machinery/rnd/server/master/attackby(obj/item/attacking_item, mob/user, params)
+	if(!source_code_hdd)
+		return ..()
+
+	switch(deconstruction_state)
+		if(HDD_PANEL_CLOSED)
+			if(istype(attacking_item, /obj/item/screwdriver/hdd_extraction))
+				to_chat(user, span_notice("Your [attacking_item.name] fits perfectly into the slots of [src]'s front panel screws. You can see [front_panel_screws] screws. You start unscrewing them..."))
+				while(attacking_item.use_tool(src, user, 100, volume=100))
+					front_panel_screws--
+
+					if(front_panel_screws <= 0)
+						deconstruction_state = HDD_PANEL_OPEN
+						to_chat(user, span_notice("You remove the last screw from [src]'s front panel. It drops, held only by wires."))
+						add_overlay("RD-server-hdd-panel-open")
+						return
+
+					to_chat(user, span_notice("You remove a screw from [src]'s front panel. Only [front_panel_screws] left..."))
+				return
+			if(istype(attacking_item, /obj/item/computer_hardware/hard_drive/cluster/hdd_theft))
+				to_chat(user, span_notice("You can't see a place to insert [src]."))
+		if(HDD_PANEL_OPEN)
+			if(istype(attacking_item, /obj/item/crowbar/hdd_extraction))
+				to_chat(user, span_notice("You can see [source_code_hdd] in a secure housing behind the front panel. Your tiny [attacking_item.name] barely fits inside. You begin to pry it loose..."))
+				if(attacking_item.use_tool(src, user, 200, volume=100))
+					to_chat(user, span_notice("You pry [source_code_hdd] free of its secure housing."))
+					deconstruction_state = HDD_PRIED
+				return
+			if(istype(attacking_item, /obj/item/computer_hardware/hard_drive/cluster/hdd_theft))
+				to_chat(user, span_notice("There's already \an [source_code_hdd.name] in the server. No need to add a second."))
+		if(HDD_PRIED)
+			if(istype(attacking_item, /obj/item/wirecutters/hdd_extraction))
+				to_chat(user, span_notice("With [source_code_hdd] free, you begin to delicately cut wires with [attacking_item]. Following the instructions on its display, there are [hdd_wires] wires left..."))
+				while(attacking_item.use_tool(src, user, 100, volume=100))
+					hdd_wires--
+
+					if(hdd_wires <= 0)
+						deconstruction_state = HDD_CUT_LOOSE
+						to_chat(user, span_notice("You cut the final wire and remove [source_code_hdd]."))
+						try_put_in_hand(source_code_hdd, user)
+						source_code_hdd = null
+						return
+
+					to_chat(user, span_notice("You cut the wire indicated by [attacking_item]. [hdd_wires] wires left..."))
+				return
+			if(istype(attacking_item, /obj/item/computer_hardware/hard_drive/cluster/hdd_theft))
+				to_chat(user, span_notice("There's already a [source_code_hdd.name] in the server. No need to add a second."))
+		if(HDD_CUT_LOOSE)
+			if(istype(attacking_item, /obj/item/computer_hardware/hard_drive/cluster/hdd_theft))
+				to_chat(user, span_notice("[src] is too badly damaged to add [attacking_item]."))
```

Comment:
```suggestion
				balloon_alert(user, "it's too badly damaged!")
```

Reply:
Should also potentially `return` here, otherwise it'll call normal attacking code and hit the thing.

Reply:
Same with all the other branches like this.

Reply:
Done for all similar branches.